### ISSUE TITLE
LightOpenGlView does not rotate with the change in screen orientation

### ISSUE THREAD

**Issue by StewardH** [pedroSG94/rtmp-rtsp-stream-client-java#1028]
While using the LightOpenGlView  the image captured by the camera  does not rotate with the change in orientation of the screen. Instead it stretches to match the new width and height of the layout.

To fix that i have replaced the call to  processMatrix and calculateViewPort in drawFrame(int width, int height, boolean keepAspectRatio, int mode, int rotation,       boolean isPreview, boolean flipStreamVertical, boolean flipStreamHorizontal) of SimpleCameraRender with

GLES20.glViewport(0, 0, width, height);

I wonder if this change compromises something else in the behavior of the  LightOpenGlView  class.

**Comment by pedroSG94** [comment]
Hello,

I'm not sure what do you mean with "no rotate with the change in orientation of the screen". My library is designed to work without rotate screen orientation. The idea is to block orientation after call startStream/startRecord like a normal camera app.

Your change affect to record video, stream and preview size so if you check all and it is working as required you can use it without problems.

**Comment by StewardH** [comment]
Hi Pedro,

Thank  you for the prompt reply. Your library is awesome. In the past I used the OpenGlView class and it was working great rotating the view while streaming when the phone orientation was changing. However I ended up needing a quick restart of the containing activity and  from time to time the OpenGlView was crashing my activity by throwing the following exceptions as I was restarting or shooting down :

E/libEGL: Could not restore original EGL context  eglMakeCurrent:891 error 3009 (EGL_BAD_MATCH)
E/SurfaceManager: eglMakeCurrent failed
E/GLConsumer: [SurfaceTexture-1-21878-409] checkAndUpdateEglState: invalid current EGLContext
I/SurfaceManager: GL released
I/Adreno: DequeueBuffer: dequeueBuffer failed
E/SurfaceManager: GL already released
E/libEGL: eglDestroySurface:663 error 300d (EGL_BAD_SURFACE)
I/SurfaceManager: GL released
E/SurfaceManager: GL already released
E/AndroidRuntime: FATAL EXCEPTION: glThread
    Process: ...: 21878
    java.lang.RuntimeException: drawFilter end. GL error: 1281
        at ....GlUtil.checkGlError(GlUtil.java:127)
        at ....gl.render.filters.BaseFilterRender.draw(BaseFilterRender.java:55)
        at ....gl.render.ManagerRender.drawOffScreen(ManagerRender.java:62)
        at ....gl.OpenGlView.run(OpenGlView.java:136)
        at java.lang.Thread.run(Thread.java:764)
E/libEGL: eglTerminate:329 error 3008 (EGL_BAD_DISPLAY)

I tried releasing all the resources before shutdown by calling the following in Camera2Base :
       stopRecord();
       stopStream();
       stopPreview();


No luck so I decided to replace the OpenGlView  with the LightOpenGlView  version. And that got rid of the  EGL_BAD_DISPLAY error above at shutdown however unlike while using OpenGlView   when relying on LightOpenGlView  the rotations were disabled at streaming as you explained above and my camera started with a 90 deg initial rotation.  I have not checked the video recording but the preview and streaming part seems to work so far with the change above. Thank you for the clarification.
